docs/admin/external-auth.md

```diff
@@ -108,6 +109,21 @@ CODER_EXTERNAL_AUTH_0_AUTH_URL="https://app.vssps.visualstudio.com/oauth2/author
 CODER_EXTERNAL_AUTH_0_TOKEN_URL="https://app.vssps.visualstudio.com/oauth2/token"
 ```
 
+### Azure DevOps (via Entra ID)
+
+Azure DevOps (via Entra ID) requires the following environment variables:
+
+```env
+CODER_EXTERNAL_AUTH_0_ID="primary-azure-devops"
+CODER_EXTERNAL_AUTH_0_TYPE=azure-devops-entra
+CODER_EXTERNAL_AUTH_0_CLIENT_ID=xxxxxx
+CODER_EXTERNAL_AUTH_0_CLIENT_SECRET=xxxxxxx
+CODER_EXTERNAL_AUTH_0_AUTH_URL="https://login.microsoftonline.com/<TENANT ID>/oauth2/authorize"
+CODER_EXTERNAL_AUTH_0_TOKEN_URL="https://login.microsoftonline.com/<TENANT ID>/oauth2/token"
+```
+
+> Note: Your app registration in Entra ID requires the `vso.code_write` scope
```

Comment:
Can we make this a default scope? I made it one in my commit.

Reply:
Not sure what you mean by this? Are you saying replace `vso.code_write` with a default scope?

Reply:
I just checked, I did not make it a default scope in my commit. Whoops!

If this scope should always be set, we can include it in the defaults.
```go
	defaults := codersdk.ExternalAuthConfig{
		DisplayName: "Azure DevOps (Entra)",
		DisplayIcon: "/icon/azure-devops.svg",
		Regex:       `^(https?://)?dev\.azure\.com(/.*)?$`,
		Scopes:      []string{"vso.code_write"},
	}
```

Reply:
Ah that's not what I mean by my comment in the docs, I mean when you create the App Registration you have to give it `vso.code_write` in the Entra ID portal
![image](https://github.com/coder/coder/assets/38583982/93f8c167-9135-4d11-8a91-d12b1eab843b)

Reply:
Oh! Understood, so there are no default scope recommendations on the Coder end?

Reply:
That's right, as I understand it we just request the 'resource' and receive a token will all scopes that have been set in the app registration